sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1452,7 +1452,10 @@ static String getExtractDestinationUri(String extractDestinationDir) {
     }
 
     @VisibleForTesting
-    Write<T> withTestServices(BigQueryServices testServices) {
+    /**
+     * This method is for test usage only
+     */
+    public Write<T> withTestServices(BigQueryServices testServices) {
```

Comment:
I think it's fine to make BigQueryServices public but add "@Experimental(Experimental.Kind.SOURCE_SINK)" so that we can remove it in the future if needed.

Reply:
You mean that I should make BigQueryServices public in addition to FakeBigQueryServices, annotate it experimental and add bigquery test artifacts only to the deps of nexmark test artifact?

Reply:
Yes. I assume "inject BigQueryServices to savePerfsToBigQuery()" is for testing. BigQueryServicesImpl should not be public. I don't think there's a point in making 'withTestServices' available for testing without making 'BigQueryServices' interface available. I think 'withTestServices' in combination with FakeBigQueryServices will be useful for other Beam components that need to test pipelines that write to BigQuery as well.

Reply:
Yes the injection is for testing only. I agree impl should not be public, and public FakeBigQueryServices, BigQueryServices interface and withTestServices will be usefull for other that have the same needs that this PR